providers/base/tests/test_wifi_nmcli_test.py

```diff
@@ -36,3 +51,563 @@ def test_legacy_nmcli_false(self, subprocess_mock):
+            "Wireless2": {"uuid": "uuid2", "state": "activated"},
+        },
+    )
+    def test_turn_down_multiple_connections(
+        self, get_connections_mock, sp_call_mock
+    ):
+        turn_down_nm_connections()
+        self.assertEqual(get_connections_mock.call_count, 1)
+        calls = [
+            call("nmcli c down uuid1".split()),
+            call("nmcli c down uuid2".split()),
+        ]
+        sp_call_mock.assert_has_calls(calls, any_order=True)
+
+
+class TestDeleteTestApSsidConnection(unittest.TestCase):
+    @patch("wifi_nmcli_test.sp.call", return_value=0)
+    @patch(
+        "wifi_nmcli_test._get_nm_wireless_connections",
+        return_value={
+            "TEST_CON": {"uuid": "uuid-test", "state": "deactivated"}
+        },
+    )
+    @patch("wifi_nmcli_test.print")
+    def test_delete_existing_test_con(
+        self, get_nm_wireless_connections_mock, sp_call_mock, print_mock
+    ):
+        delete_test_ap_ssid_connection()
+
+    @patch("wifi_nmcli_test.sp.call", side_effect=Exception("Deletion failed"))
+    @patch(
+        "wifi_nmcli_test._get_nm_wireless_connections",
+        return_value={
+            "TEST_CON": {"uuid": "uuid-test", "state": "deactivated"}
+        },
+    )
+    def test_delete_test_con_exception(
+        self, get_nm_wireless_connections_mock, sp_call_mock
+    ):
+        delete_test_ap_ssid_connection()
+
+    @patch("wifi_nmcli_test._get_nm_wireless_connections", return_value={})
+    def test_no_test_con_to_delete(self, get_nm_wireless_connections_mock):
+        delete_test_ap_ssid_connection()
+
+
+class TestListAps(unittest.TestCase):
+    @patch("wifi_nmcli_test.sp.check_output")
+    def test_list_aps_no_essid(self, check_output_mock):
+        check_output_mock.return_value = (
+            b"wlan0 \nSSID1:1:2412:60\nSSID2:6:2437:70\nSSID3:11:2462:80"
+        )
+        expected = {
+            "SSID1": {"Chan": "1", "Freq": "2412", "Signal": "60"},
+            "SSID2": {"Chan": "6", "Freq": "2437", "Signal": "70"},
+            "SSID3": {"Chan": "11", "Freq": "2462", "Signal": "80"},
+        }
+        self.assertEqual(list_aps("wlan0"), expected)
+
+    @patch("wifi_nmcli_test.sp.check_output")
+    def test_list_aps_with_essid(self, check_output_mock):
+        check_output_mock.return_value = (
+            b"SSID1:1:2412:60\nSSID2:6:2437:70\nSSID3:11:2462:80"
+        )
+        expected = {
+            "SSID2": {"Chan": "6", "Freq": "2437", "Signal": "70"},
+        }
+        self.assertEqual(list_aps("wlan0", "SSID2"), expected)
+
+    @patch("wifi_nmcli_test.sp.check_output")
+    def test_list_aps_empty_output(self, check_output_mock):
+        check_output_mock.return_value = b""
+        expected = {}
+        self.assertEqual(list_aps("wlan0"), expected)
+
+
+class TestShowAps(unittest.TestCase):
+    @patch("wifi_nmcli_test.print")
+    def test_show_aps_empty(self, mock_print):
+        aps_dict = {}
+        show_aps(aps_dict)
```

Comment:
Same, check that the correct message is printed

Reply:
Fixed